core/iwasm/interpreter/wasm_runtime.c

```diff
@@ -2157,7 +2157,6 @@ wasm_call_function(WASMExecEnv *exec_env, WASMFunctionInstance *function,
     wasm_exec_env_set_thread_info(exec_env);
 
     interp_call_wasm(module_inst, exec_env, function, argc, argv);
-    (void)clear_wasi_proc_exit_exception(module_inst);
```

Comment:
Not sure why remove this line?

The wasi proc_exit native API is to terminate the process normally, but if runtime exits directly in the native API wasi_proc_exit of libc-wasi, some resources won't be destroyed, and there might be other threads running with the same module instance, normally runtime should wait for all threads of the module instance to terminate before destroy the module instance. So currently runtime just throws exception "wasi proc exit" in the native API, and clear this exception after call wasm funtion to make it exit normally.

For example, when using emsdk to build a "hello world" wasm app, normally it calls proc_exit at the end. If we don't throw "wasi proc exit" exception in the native API, then the wasm app will continue to execute the successive opcodes and "unreachable" exception will be thrown instead.

And if we don't clear "wasi_proc_exception", it will be thrown.

Reply:
I was commenting it so that after calling `wasm_runtime_call_wasm` we can check if a `proc_exit` occurred to propagate it; since with wasi threads we have one instance per thread, do you think it would be enough to just wrap the changes with `WASM_ENABLE_LIB_WASI_THREADS`?
If not, what's your suggestion? I started contributing to this project not long ago so any clarification/suggestion is much appreciated

Reply:
There are some places calling clear_wasi_proc_exit_exception, I think if you need to disable the clear exception action, had better just modify the clear_wasi_proc_exit_exception itself, in both wasm_runtime.c and aot_runtime:

https://github.com/bytecodealliance/wasm-micro-runtime/blob/main/core/iwasm/interpreter/wasm_runtime.c#L2037

https://github.com/bytecodealliance/wasm-micro-runtime/blob/main/core/iwasm/aot/aot_runtime.c#L903